File: codegen/results/v002/ollama/qwen3.5_27b-nvfp4_think/20260412_055759/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

# Reference date for age calculation: July 1, 2025
reference_date = date(2025, 7, 1)

with open('input/input.csv', 'r') as f:
    reader = csv.DictReader(f)
    result = []
    
    for row in reader:
        # Parse name into first and last
        parts = row['Name'].split()
        first_name = parts[0]
        last_name = parts[-1]
        
        # Parse birthday from M/D/YYYY format
        bday_parts = row['Birthday'].split('/')
        birth_month, birth_day, birth_year = int(bday_parts[0]), int(bday_parts[1]), int(bday_parts[2])
        
        # Calculate age as of July 1, 2025
        age = reference_date.year - birth_year
        if (reference_date.month, reference_date.day) < (birth_month, birth_day):
            age -= 1
        
        # Format birthday in ISO format (YYYY-MM-DD)
        birthday_iso = f"{birth_year}-{birth_month:02d}-{birth_day:02d}"
        
        # Build relatives list from Father, Mother, Brother, Sister columns
        relatives = []
        
        for rel_field, rel_type in [('Father', 'Father'), ('Mother', 'Mother'), 
                                     ('Brother', 'Brother'), ('Sister', 'Sister')]:
            value = row.get(rel_field, '').strip()
            if value and value.lower() != 'null':
                name_parts = value.split()
                relatives.append({
                    'FirstName': name_parts[0],
                    'LastName': name_parts[-1],
                    'Relationship': rel_type
                })
        
        result.append({
            'FirstName': first_name,
            'LastName': last_name,
            'Birthday': birthday_iso,
            'Age': age,
            'Relatives': relatives
        })

# Output only the JSON array to stdout
print(json.dumps(result))

File: codegen/results/v002/ollama/qwen3.5_27b-nvfp4_think/20260412_055759/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface Person {
  FirstName: string;
  LastName: string;
  Birthday: string;
  Age: number;
  Relatives: Relative[];
}

interface Relative {
  FirstName: string;
  LastName: string;
  Relationship: string;
}

function parseName(name: string): { firstName: string; lastName: string } {
  const parts = name.trim().split(/\s+/);
  if (parts.length === 1) {
    return { firstName: parts[0], lastName: '' };
  }
  return {
    firstName: parts[0],
    lastName: parts.slice(1).join(' ')
  };
}

function parseDate(dateStr: string): Date | null {
  if (!dateStr || dateStr.toLowerCase() === 'null') return null;
  
  const match = dateStr.match(/^(\d+)\/(\d+)\/(\d+)$/);
  if (!match) return null;
  
  const [, month, day, year] = match.map(Number);
  const date = new Date(year, month - 1, day);
  return isNaN(date.getTime()) ? null : date;
}

function formatBirthday(date: Date): string {
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  return `${year}-${month}-${day}`;
}

function calculateAge(birthday: Date, deathDate: Date | null): number {
  const referenceDate = new Date(2025, 6, 1); // July 1, 2025
  
  let age = referenceDate.getFullYear() - birthday.getFullYear();
  
  const birthdayThisYear = new Date(referenceDate.getFullYear(), birthday.getMonth(), birthday.getDate());
  if (referenceDate < birthdayThisYear) {
    age--;
  }
  
  if (deathDate && deathDate < referenceDate) {
    let deathAge = deathDate.getFullYear() - birthday.getFullYear();
    const birthdayAtDeath = new Date(deathDate.getFullYear(), birthday.getMonth(), birthday.getDate());
    if (deathDate < birthdayAtDeath) {
      deathAge--;
    }
    return Math.max(0, deathAge);
  }
  
  return age;
}

function main() {
  const inputPath = path.join('input', 'input.csv');
  const csvContent = fs.readFileSync(inputPath, 'utf-8');
  
  const lines = csvContent.trim().split('\n');
  const headers = lines[0].split(',');
  
  const results: Person[] = [];
  
  for (let i = 1; i < lines.length; i++) {
    const values = lines[i].split(',');
    const row: Record<string, string> = {};
    
    headers.forEach((header, index) => {
      row[header] = values[index] || '';
    });

    const nameParts = parseName(row['Name']);
    const birthday = parseDate(row['Birthday']);
    const deathDate = parseDate(row['Died']);
    const age = birthday ? calculateAge(birthday, deathDate) : 0;

    const relatives: Relative[] = [];
    
    if (row['Father'] && row['Father'].toLowerCase() !== 'null') {
      const relative = parseName(row['Father']);
      relatives.push({
        FirstName: relative.firstName,
        LastName: relative.lastName,
        Relationship: 'Father'
      });
    }
    
    if (row['Mother'] && row['Mother'].toLowerCase() !== 'null') {
      const relative = parseName(row['Mother']);
      relatives.push({
        FirstName: relative.firstName,
        LastName: relative.lastName,
        Relationship: 'Mother'
      });
    }
    
    if (row['Brother'] && row['Brother'].toLowerCase() !== 'null') {
      const relative = parseName(row['Brother']);
      relatives.push({
        FirstName: relative.firstName,
        LastName: relative.lastName,
        Relationship: 'Brother'
      });
    }
    
    if (row['Sister'] && row['Sister'].toLowerCase() !== 'null') {
      const relative = parseName(row['Sister']);
      relatives.push({
        FirstName: relative.firstName,
        LastName: relative.lastName,
        Relationship: 'Sister'
      });
    }
    
    results.push({
      FirstName: nameParts.firstName,
      LastName: nameParts.lastName,
      Birthday: birthday ? formatBirthday(birthday) : '',
      Age: age,
      Relatives: relatives
    });
  }

  console.log(JSON.stringify(results));
}

main();